From 7ecfcb5fe0fad80c46420609a818b8139474f39e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Fri, 12 Feb 2016 17:46:34 +0200
Subject: [PATCH 1137/1429] crypto: caam - add ARCH_LAYERSCAPE to supported
 architectures
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This basically adds support for ls1043a platform.

Note: this works as far as ARCH_LAYERSCAPE platforms like ls2080a
won't use this driver, since:
-ls1043a: CAAM is big endian
-ls2080a: CAAM is little endian
and for this to work we either need a separate defconfig for the
two platforms or support for endianness detection at runtime.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/crypto/caam/Kconfig |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/Kconfig b/drivers/crypto/caam/Kconfig
index 69f1a80..ca8873c 100644
--- a/drivers/crypto/caam/Kconfig
+++ b/drivers/crypto/caam/Kconfig
@@ -1,6 +1,6 @@
 config CRYPTO_DEV_FSL_CAAM
 	tristate "Freescale CAAM-Multicore driver backend"
-	depends on FSL_SOC || ARCH_MXC
+	depends on FSL_SOC || ARCH_MXC || ARCH_LAYERSCAPE
 	help
 	  Enables the driver module for Freescale's Cryptographic Accelerator
 	  and Assurance Module (CAAM), also known as the SEC version 4 (SEC4).
@@ -148,7 +148,7 @@ config CRYPTO_DEV_FSL_CAAM_IMX
 	depends on CRYPTO_DEV_FSL_CAAM
 
 config CRYPTO_DEV_FSL_CAAM_LE
-	def_bool CRYPTO_DEV_FSL_CAAM_IMX || SOC_LS1021A
+	def_bool (CRYPTO_DEV_FSL_CAAM_IMX || SOC_LS1021A) && !ARCH_LAYERSCAPE
 	depends on CRYPTO_DEV_FSL_CAAM
 
 config CRYPTO_DEV_FSL_CAAM_DEBUG
-- 
1.7.5.4

